hvplot/converter.py

```diff
@@ -571,27 +572,29 @@ def _process_data(self, kind, data, x, y, by, groupby, row, col,
         self.hover_cols = hover_cols
         self._redim_label = {}
 
-        if da is not None:
+        if da is not None and attr_labels:
             try:
-                xyz_tuples = [('x', x, da[x].attrs), ('y', y, da[y].attrs)]
-                xyz_tuples.append(('z', da.name, da.attrs))
-            except ValueError:  # simple plot
-                xyz_tuples = [('x', x, da[x].attrs), ('y', da.name, da.attrs)]
-
-            long_name = {}
-            units = {}
-            # var as in variable name
-            for xyz_str, xyz_name, xyz_attrs in xyz_tuples:
-                long_name[xyz_str] = xyz_attrs.get('long_name', '')
-                if 'units' in xyz_attrs:
-                    units[xyz_str] = '[{0}]'.format(
-                        xyz_attrs['units'])
-                else:
-                    units[xyz_str] = ''
-                xyz_label = ' '.join([long_name[xyz_str],
-                                      units[xyz_str]])
-                if xyz_label.strip() != '':
-                    self._redim_label[xyz_name] = xyz_label
+                xyz_tuples = [(i, z, da[z].attrs) for i, z in enumerate(da.coords)]
+                xyz_tuples.append(('v', da.name, da.attrs))
+
+                long_name = {}
+                units = {}
+                # var as in variable name
+                for xyz_str, xyz_name, xyz_attrs in xyz_tuples:
+                    long_name[xyz_str] = xyz_attrs.get('long_name', '')
+                    if 'units' in xyz_attrs:
+                        units[xyz_str] = '[{0}]'.format(
+                            xyz_attrs['units'])
+                    else:
+                        units[xyz_str] = ''
+                    xyz_label = ' '.join([long_name[xyz_str],
+                                          units[xyz_str]])
```

Comment:
Instead of joining the label and unit they should be separate attributes, so instead of setting ``_redim_label`` you should accumulate into ``_redim``.